lib/usage-tracking/class-usage-tracking.php

```diff
@@ -0,0 +1,454 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Usage Tracking class. Please update the prefix to something unique to your
+ * plugin.
+ */
+class Sensei_Usage_Tracking {
+
+	/****** Plugin-specific section ******/
```

Comment:
I still think it is a good idea to eventually move the plugin-specific stuff to its own project-specific class (in `includes`) that extends this class and make this an abstract class. You'll need to shift `private` access to `protected`.

Reply:
Done. Base implementation and tests are in `lib/usage-tracking`. Plugin-specific subclass is in `includes/class-sensei-usage-tracking.php`